modules/base/rendering/pointcloud/renderablepolygoncloud.cpp

```diff
@@ -0,0 +1,190 @@
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,   *
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A         *
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT    *
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF  *
+ * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE  *
+ * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                                         *
+ ****************************************************************************************/
+
+#include <modules/base/rendering/pointcloud/renderablepolygoncloud.h>
+
+#include <openspace/documentation/documentation.h>
+#include <ghoul/filesystem/filesystem.h>
+#include <ghoul/glm.h>
+#include <ghoul/logging/logmanager.h>
+#include <ghoul/opengl/programobject.h>
+#include <ghoul/opengl/texture.h>
+#include <optional>
+
+namespace {
+    constexpr std::string_view _loggerCat = "RenderablePolygonCloud";
+
+    // A RenderablePolygonCloud is a RenderablePointCloud where the shape of the points
+    // is a uniform polygon with a given number of sides instead of a texture. For
+    // instance, PolygonSides = 5 results in the points being rendered as pentagons.
+    // Note that while this renderable inherits the texture property from
+    // RenderablePointCloudany, any added texture value will be ignored in favor of the
+    // polygon shape.
+    //
+    // Se documentation of RenderablePointCloud for details on the other parts of the
+    // point cloud rendering.
+    struct [[codegen::Dictionary(RenderablePolygonCloud)]] Parameters {
+        // The number of sides for the polygon used to represent each point. Default is
+        // 3, i.e. to use triangles
+        std::optional<int> polygonSides;
+    };
+
+#include "renderablepolygoncloud_codegen.cpp"
+}  // namespace
+
+namespace openspace {
+
+documentation::Documentation RenderablePolygonCloud::Documentation() {
+    return codegen::doc<Parameters>(
+        "base_renderablepolygoncloud",
+        RenderablePointCloud::Documentation()
+    );
+}
+
+RenderablePolygonCloud::RenderablePolygonCloud(const ghoul::Dictionary& dictionary)
+    : RenderablePointCloud(dictionary)
+{
+    const Parameters p = codegen::bake<Parameters>(dictionary);
+
+    _polygonSides = p.polygonSides.value_or(_polygonSides);
+}
+
+void RenderablePolygonCloud::initializeGL() {
+    ZoneScoped;
+
+    RenderablePointCloud::initializeGL();
+    createPolygonTexture();
+}
+
+void RenderablePolygonCloud::deinitializeGL() {
+    glDeleteBuffers(1, &_polygonVbo);
+    _polygonVbo = 0;
+    glDeleteVertexArrays(1, &_polygonVao);
+    _polygonVao = 0;
+
+    glDeleteTextures(1, &_pTexture);
+}
+
+void RenderablePolygonCloud::bindTextureForRendering() const {
+    glBindTexture(GL_TEXTURE_2D, _pTexture);
+}
+
+void RenderablePolygonCloud::createPolygonTexture() {
```

Comment:
This depends on what kind of resources we prefer to prioritize, I guess. This textured-based approach is how it was implemented before the rewrite and it seems to make sense to me

The way it works now, the texture is created once (in the initializeGl) and then just used for rendering. If we do it in the geometry shader, that would add extra logic in the shader itself and we would also have to generate more vertices every time the point is rendered (currently, we only need 4, for the texture plane, and this is the same for all the use cases)

I don't know which approach is preferable, but  I like that it does not lead to another path or any special check in the geomoetry shader code that would be special for polygons. But I'm open to suggestions if we think that approach is better

What are your thoughts, @alexanderbock and @Roxeena ?

Reply:
IIrc, we experimented with Jonathas back when and found that the texture-based approach was running faster than the geometry shader, since we are doing it for a lot of points and if the vertex shader places a point outside the view frustum it might get culled away early, but if a geometry shader is running it can only be culled afterwards since the shader might move it _back_ into the frustum.  So a lot of extra work that impacted performance

Reply:
Hm, ok.  Although, there is a geometry shader running anyways, to generate the texture plane, so I'm not sure I understand how it would be any different 🤔 

Either way, I don't really think we need to update the implementation

Reply:
I don't have any strong feelings, it just looked like we had two different solutions to the same problem but they are doing different things I suppose.